src/go/serviceconfig/config_id_fetcher.go

```diff
@@ -0,0 +1,71 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package serviceconfig
+
+import (
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/GoogleCloudPlatform/esp-v2/src/go/util"
+	"github.com/golang/protobuf/proto"
+	scpb "google.golang.org/genproto/googleapis/api/servicecontrol/v1"
+)
+
+type ConfigIdFetcher struct {
```

Comment:
I prefer `ServiceConfigIdFetcher`. Config sounds too general, we have many configs in our code :)